scripts/mk_all.sh

```diff
@@ -1,6 +1,33 @@
 #! /usr/bin/env bash
+
+##  Standard use:
+##  ./scripts/mk_all.sh
+##
+##  updates `Mathlib` `MathlibExtras` `Mathlib/Tactic` `Counterexamples` `Archive`
+##
+##  Pro use:
+##  ./scripts/mk_all.sh <space_separated_list_of_dirs>
+##
+##  for each dir in <space_separated_list_of_dirs>, create a file with all the imports of the dir
+##
+##  The two commands
+##  ./scripts/mk_all.sh
+##  ./scripts/mk_all.sh Mathlib MathlibExtras Mathlib/Tactic Counterexamples Archive
+##
+##  are equivalent.
+
 cd "$(git rev-parse --show-toplevel)" || exit 1
-for gp in Mathlib MathlibExtras Mathlib/Tactic Counterexamples Archive
+
+# assign `targets` to be the input if provided, default to "the usual Mathlib suspects" otherwise
+if [ -n "${*}" ]
+then
+  targets=${*}
```

Comment:
I don't know enough bash to review this, but would approve the CI changes.

Reply:
If it helps, `-n` checks whether the variable is assigned and non empty. `$*` means "all arguments passed to the function".

Reply:
I've looked at this enough to convince myself this should do what it says.
In any case, #11874 offers rewriting this in Lean - so I am not worried about long-term maintainability. Short-term, this seems fine and like a clear step forward.

Reply:
Thinking about this again, I don't see the use case for passing arguments to this script. Unless there's a strong reason for it that I haven't thought about, I'd revert that part. The main selling point of this script is that it's simple and fast, IMO.

Reply:
The reason I added that option is for downstream projects that can target whichever folder they want.  This was before the lean script, though.